### PR TITLE
enh: add network policy for core and fix it for oauth

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- `oauth` policy is wrong (`front` and `connectors` should be allowed, but not `core`)
- add network policy for `core` (should only be reachable from `front` or `front-worker`)

Should we have `prodbox` as well ? 🤔 

## Risk

Should be fine, but if something is wrong we can't reach `core` anymore

## Deploy Plan

apply infra

### CODE REVIEW

File: k8s/network-policies/core-network-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: oauth-network-policy
```
Comment (flvndvd) [inline k8s/network-policies/core-network-policy.yaml:4]:
```suggestion
  name: core-network-policy
```

**Review by flvndvd — APPROVED** [review]
LGTM

File: k8s/network-policies/core-network-policy.yaml (hunk 1 above)
Comment (fontanierh) [reply k8s/network-policies/core-network-policy.yaml:4]: Good catch thank you

**Comment by spolu** [comment]
Very naice

**Pull request merged**